auto_gptq/nn_modules/qlinear/qlinear_cuda_old.py

```diff
@@ -229,44 +231,45 @@ def forward(self, x):
                     raise NotImplementedError("Only 2,3,4,8 bits are supported.")
         else:
             if self.wf.device != self.qzeros.device:
-               self.wf = self.wf.to(self.qzeros.device)
+                self.wf = self.wf.to(self.qzeros.device)
                 
             if self.bits in [2,4,8]:
-               zeros = torch.bitwise_right_shift(torch.unsqueeze(self.qzeros, 2).expand(-1, -1, 32 // self.bits), self.wf.unsqueeze(0)).to(torch.int16 if self.bits == 8 else torch.int8)
-               zeros = torch.bitwise_and(zeros, (2 ** self.bits) - 1)
-                   
-               zeros = zeros + 1
-               zeros = zeros.reshape(-1, 1, zeros.shape[1] * zeros.shape[2])
-   
-               scales = self.scales
-               scales = scales.reshape(-1, 1, scales.shape[-1])
+                zeros = torch.bitwise_right_shift(torch.unsqueeze(self.qzeros, 2).expand(-1, -1, 32 // self.bits), self.wf.unsqueeze(0)).to(torch.int16 if self.bits == 8 else torch.int8)
+                
+                zeros = zeros + 1
+                zeros = torch.bitwise_and(zeros, (2 ** self.bits) - 1)  # NOTE: It appears that casting here after the `zeros = zeros + 1` is important.
```

Comment:
Swapping these two ops was particularly important to maintain quality cc @wejoncy I saw you did the same, did you observe the same thing?

Right now the exllama/cuda-old CUDA kernels are outputting trash due IMO to not checking that we don't overflow on int4.

Reply:
Yes,  so that's why I removed `zeros = zeros - 1` in quant process.
`zeros` ranges from 0-15
`zeros-1` ranges from -1-14 where -1 will be represented as 255 as we don't have data structure int4
plus `torch.bitwise_and`   (15+1)&15 =0,  instead of 16 where we represent it in int32 during computing.

Reply:
Can AutoGPTQ also keep zeros as original intead of zeros-1? I didn't find any drawbacks yet.